src/python/gudhi/simplex_tree.pyx

```diff
@@ -228,6 +237,81 @@ cdef class SimplexTree:
         """
         return self.get_ptr().insert(simplex, <double>filtration)
 
+    @staticmethod
+    @cython.boundscheck(False)
+    def create_from_array(filtrations, double max_filtration=np.inf):
+        """Creates a new, empty complex and inserts vertices and edges. The vertices are numbered from 0 to n-1, and
+        the filtration values are encoded in the array, with the diagonal representing the vertices. It is the
+        caller's responsibility to ensure that this defines a filtration, which can be achieved with either::
+
+            filtrations[np.diag_indices_from(filtrations)] = filtrations.min(1)
+
+        or::
+
+            diag = filtrations.diagonal()
+            filtrations = np.fmax(np.fmax(filtrations, diag[:, None]), diag[None, :])
+
+        :param filtrations: the filtration values of the vertices and edges to insert. The matrix is assumed to be symmetric.
+        :type filtrations: numpy.ndarray of shape (n,n)
+        :param max_filtration: only insert vertices and edges with filtration values no larger than max_filtration
+        :type max_filtration: float
+        :returns: the new complex
+        :rtype: SimplexTree
+        """
+        # TODO: document which half of the matrix is actually read?
+        filtrations = np.asanyarray(filtrations, dtype=float)
+        cdef double[:,:] F = filtrations
+        ret = SimplexTree()
+        cdef int n = F.shape[0]
+        assert n == F.shape[1]
+        with nogil:
+            ret.get_ptr().insert_matrix(&F[0,0], n, F.strides[0], F.strides[1], max_filtration)
+        return ret
+
+    def insert_edges_from_coo_matrix(self, edges):
```

Comment:
Just a very naive question, but why not a static `create_from_coo_matrix` method instead ? (I understand the point that we do not want to insert the zeros from the sparse matrix)

Reply:
I think we already discussed it a bit in https://github.com/GUDHI/gudhi-devel/pull/331#issuecomment-668584741
This may be more obvious from the name for the function `insert_batch`, but it can be useful to insert a batch of edges in a complex that already has vertices (or possibly even other simplices).
If we have insert_batch for that, we could reserve the one with a coo_matrix for creation only, and people can still add vertices afterwards if they want, even if that's a little awkward. Or, as I mentioned, we could just remove this function and consider that insert_batch is enough. Or we could have both the insert_ and create_ versions. I don't have a very strong opinion there.